### PR TITLE
feat: error/warn on unsupported browser versions

### PR DESCRIPTION
**Pull request by flotwig**
- Closes 

### User facing changelog

- Cypress now enforces version checks for browser launching. Cypress supports Chrome >= 64, Firefox >= 86, and Edge >= 79.

### How has the user experience changed?

In `cypress open`, an unsupported browser will appear in the drop down, but greyed out and with a warning tooltip. 

If the user clicks it, nothing will happen - the dropdown remains open and the unsupported browser is not selected.

If the user has an unsupported browser stored in their `localStorage` as the most recently chosen browser, the first supported detected browser will be selected instead next time the desktop GUI is opened.

image

In `cypress run`, an error is printed and the process exits:

image

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/launcher/lib/detect.ts [hunk 1]
```diff
@@ -16,26 +16,26 @@ import {
...
-      'browser %s version %s major version %s',
-      browser.name,
-      browser.version,
-      majorVersion,
-    )
-  }
+  const majorVersion = parseInt(browser.version.split('.')[0]) || browser.version
+
+  const unsupportedVersion = browser.minSupportedVersion && majorVersion < browser.minSupportedVersion
+  const warning = unsupportedVersion ? `Cypress does not support running ${browser.displayName} version ${majorVersion} - it is too old. To use ${browser.displayName} with Cypress, install a version of ${browser.displayName} newer than or equal to ${browser.minSupportedVersion}.` : undefined
```
Comment (jennifer-shehane) [inline packages/launcher/lib/detect.ts:28]: "it is too old!!" haha, this phrase made me laugh. This phrase probably unnecessary and may be too broad in explaining why we may not support a certain browser. 

```suggestion
  const warning = unsupportedVersion ? `Cypress does not support running ${browser.displayName} version ${majorVersion}. To use ${browser.displayName} with Cypress, install a version of ${browser.displayName} newer than or equal to ${browser.minSupportedVersion}.` : undefined
```

File: packages/launcher/test/unit/detect_spec.ts [hunk 2]
```diff
@@ -151,7 +176,8 @@ describe('browser detection', () => {
       execa.withArgs('/bad-firefox', ['--version'])
       .resolves({ stdout: 'Mozilla Firefox 79.1' })
 
-      expect(await detectByPath('/good-firefox')).to.not.have.property('warning')
+      // TODO: remove this warning/auto-GC, since Cy no longer supports FF < 86
```
Comment (flotwig) [inline packages/launcher/test/unit/detect_spec.ts:179]: I will open a separate PR to remove the auto-GC code.

File: packages/launcher/lib/detect.ts (hunk 1 above)
Comment (flotwig) [reply packages/launcher/lib/detect.ts]: 3c6d6d3

File: packages/server/lib/project-base.ts [hunk 3]
```diff
@@ -38,8 +38,8 @@ import { checkSupportFile } from './project_utils'
...
 // TODO: Figure out how to type this better.
 type ReceivedCypressOptions =
-  Partial<Pick<Cypress.RuntimeConfigOptions, 'hosts' | 'projectName' | 'clientRoute' | 'devServerPublicPathRoute' | 'supportFolder' | 'namespace' | 'report' | 'socketIoCookie' | 'configFile' | 'isTextTerminal' | 'isNewProject' | 'proxyUrl' | 'browsers'>>
-  & Partial<Pick<Cypress.ResolvedConfigOptions, 'experimentalSourceRewriting' | 'fixturesFolder' | 'reporter' | 'reporterOptions' | 'screenshotsFolder' | 'pluginsFile' | 'supportFile' | 'integrationFolder' | 'baseUrl' | 'viewportHeight' | 'viewportWidth' | 'port' | 'experimentalInteractiveRunEvents'>>
+  Partial<Pick<Cypress.RuntimeConfigOptions, 'hosts' | 'projectName' | 'clientRoute' | 'devServerPublicPathRoute' | 'namespace' | 'report' | 'socketIoCookie' | 'configFile' | 'isTextTerminal' | 'isNewProject' | 'proxyUrl' | 'browsers'>>
+  & Partial<Pick<Cypress.ResolvedConfigOptions, 'supportFolder' | 'experimentalSourceRewriting' | 'fixturesFolder' | 'reporter' | 'reporterOptions' | 'screenshotsFolder' | 'pluginsFile' | 'supportFile' | 'integrationFolder' | 'baseUrl' | 'viewportHeight' | 'viewportWidth' | 'port' | 'experimentalInteractiveRunEvents'>>
```
Comment (flotwig) [inline packages/server/lib/project-base.ts:42]: this is failing in 8.0-release, i think this is the correct fix?

File: packages/launcher/test/unit/detect_spec.ts (hunk 2 above)
Comment (flotwig) [reply packages/launcher/test/unit/detect_spec.ts:179]: https://github.com/cypress-io/cypress/pull/17368

**Review by chrisbreiding — APPROVED** [review]
LGTM 👍

**Pull request merged**